pshtt.py

```diff
@@ -262,6 +267,10 @@ def hsts_check(endpoint):
     if "max-age" in header.lower():
         endpoint.hsts_max_age = int(temp[0][len("max-age="):])
 
+    if endpoint.hsts_max_age <= 0:
+        endpoint.hsts = False
+        return
```

Comment:
If an HSTS `max-age` is 0, this is the mechanism in the HSTS spec for disabling HSTS. Clients are supposed to clear their HSTS status for a site if they see `max-age=0`. I don't know offhand what a negative `max-age` means, but it would be nonsensical enough that I'm comfortable also saying that HSTS isn't present if that's the case.